test/hotspot/jtreg/serviceability/dcmd/thread/PrintMountedVirtualThread.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.dcmd.CommandExecutor;
+import jdk.test.lib.dcmd.JMXExecutor;
+import jdk.test.lib.process.OutputAnalyzer;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Pattern;
+
+/*
+ * @test
+ * @summary Test of diagnostic command Thread.print with virtual threads
+ * @requires vm.continuations
+ * @library /test/lib
+ * @modules java.base
+ * @run testng PrintMountedVirtualThread
+ */
+public class PrintMountedVirtualThread {
+
+    public void run(CommandExecutor executor) throws InterruptedException {
+        var shouldFinish = new AtomicBoolean(false);
+        var started = new CountDownLatch(1);
+        final Runnable runnable = new DummyRunnable(shouldFinish, started);
+        Thread vthread = Thread.ofVirtual().name("Dummy Vthread").start(runnable);
+        started.await();
+        /* Execute */
+        OutputAnalyzer output = executor.execute("Thread.print");
+        output.shouldMatch(".*at " + Pattern.quote(DummyRunnable.class.getName()) + "\\.run.*");
+        output.shouldMatch(".*at " + Pattern.quote(DummyRunnable.class.getName()) + "\\.compute.*");
+        output.shouldMatch("Mounted virtual thread " + "\"Dummy Vthread\"" + " #" + vthread.threadId());
+        shouldFinish.set(true);
```

Comment:
One other suggestion is to use a try-finally block here. Put L48-53 in the block and set shouldFinish in the finally block. That way if the test fails then it won't leave a spinning thread to disrupt the next test that runs in the agent VM.

Also just to say that we've mostly used JUnit for new tests in recent releases, moving away from TestNG for new tests.

Reply:
Thanks @AlanBateman ! In commit https://github.com/openjdk/jdk/pull/19482/commits/05d861cb38cc061c79c9b7f451bf096b1cbb0ddd I've incorporated your suggestions.